src/components/table/OcTable.vue

```diff
@@ -1,68 +1,210 @@
+            </span>
+            <span
+              class="oc-ml-xs oc-icon-l"
+              :style="[resultArray[index].open ? {} : { display: 'none' }]"
+            >
+              <oc-icon
+                name="expand_more"
+                class="oc-accordion-title-arrow-icon"
+                :class="{ rotate: resultArray[index].open }"
+                size="large"
+              /> </span></oc-td
+        ></oc-tr>
+        <template v-if="resultArray[index].open">
+          <oc-tr
+            v-for="(item, trIndex) in item.data"
+            :key="`oc-tbody-tr-${item[idKey] || trIndex}`"
+            :ref="`row-${trIndex}`"
+            v-bind="extractTbodyTrProps(item, trIndex)"
+            @click.native="$emit(constants.EVENT_TROW_CLICKED, item)"
+            @hook:mounted="$emit(constants.EVENT_TROW_MOUNTED, item, $refs[`row-${trIndex}`][0])"
+          >
+            <oc-td
+              v-for="(field, tdIndex) in fields"
+              :key="'oc-tbody-td-' + cellKey(field, tdIndex, item)"
+              v-bind="extractTdProps(field, tdIndex)"
+            >
+              <slot v-if="isFieldTypeSlot(field)" :name="field.name" :item="item" />
+              <template v-else-if="isFieldTypeCallback(field)">
+                {{ field.callback(item[field.name]) }}
+              </template>
+              <template v-else>{{ item[field.name] }}</template>
+            </oc-td>
+          </oc-tr>
+        </template>
+      </tbody>
+
+      <!-- Show more/show less footer for preview enabled -->
+      <tbody
+        v-if="
+          groupingSettings &&
+          groupingSettings.previewAmount &&
+          data.length > groupingSettings.previewAmount + 1
+        "
+      >
+        <oc-tr
+          style="
+             {
+              height: rowHeight + 'px';
+            }
+          "
+          :class="['oc-tbody-tr', 'preview-settings']"
+          @click.native="previewEnabled = !previewEnabled"
         >
-          <slot v-if="isFieldTypeSlot(field)" :name="field.name" :item="item" />
-          <template v-else-if="isFieldTypeCallback(field)">
-            {{ field.callback(item[field.name]) }}
-          </template>
-          <template v-else>{{ item[field.name] }}</template>
-        </oc-td>
-      </oc-tr>
-    </oc-tbody>
-    <tfoot v-if="$slots.footer" class="oc-table-footer">
-      <tr>
-        <td :colspan="footerColspan" class="oc-table-footer-cell">
-          <!-- @slot Footer of the table -->
-          <slot name="footer" />
-        </td>
-      </tr>
-    </tfoot>
-  </table>
+          <oc-td
+            v-if="previewEnabled"
+            key="showMore"
+            :colspan="fields.length"
+            style="
+               {
+                text-align: center;
+              }
+            "
+            ><div class="preview-heading">Show more</div> </oc-td
+          ><oc-td
+            v-else
+            key="showLess"
+            :colspan="fields.length"
+            style="
```

Comment:
One thing I forgot to mention before: can't we put the styles in the styles file instead of inline? I might not want this to be centered.